src/ansible_compat/runtime.py

```diff
@@ -113,16 +119,29 @@ def clean(self) -> None:
         if self.cache_dir:
             shutil.rmtree(self.cache_dir, ignore_errors=True)
 
-    # pylint: disable=no-self-use
-    def exec(self, args: Union[str, List[str]]) -> CompletedProcess:
-        """Execute a command inside an Ansible environment."""
-        return subprocess.run(
-            args,
-            universal_newlines=True,
-            check=False,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
+    def exec(
+        self, args: Union[str, List[str]], retry: bool = False
+    ) -> CompletedProcess:
+        """Execute a command inside an Ansible environment.
+
+        :param retry: Retry network operations on failures.
+        """
+        for _ in range(self.max_retries if retry else 1):
```

Comment:
Why not
```suggestion
        for _ in range(self.max_retries if self.retry else 1):
```

Reply:
You indeed spot an design issue but that is not really the correct fix. The default retry value used by exec should be self.retry but we should still keep it. Users may want to alter the retry behavior for specific commands, even inside the library itself we do this. Mainly galaxy commands should use retry but other operations should not.

Let's check the use cases to get it right from start.